service/k8sclient/kubectl/kubectl.go

```diff
@@ -0,0 +1,126 @@
+// dbaas-controller
+// Copyright (C) 2020 Percona LLC
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+// Package kubectl provides kubectl client.
+package kubectl
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"os/exec"
+	"strings"
+	"sync"
+
+	_ "github.com/percona/percona-server-mongodb-operator/pkg/apis/psmdb/v1"
```

Comment:
I added this page not to break build (that wasn't intentional), but to show that methods like `Apply` can't accept `*pxc.PerconaXtraDBCluster` – there will be MongoDB cluster too, as well as backups and restores CRs. So methods should be renamed to be more specific in several places.